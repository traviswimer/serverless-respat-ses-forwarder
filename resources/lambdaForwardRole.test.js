const lambdaForwardRole = require('./lambdaForwardRole');

describe('lambdaForwardRole()', () => {
	test('generates role', () => {
		expect(lambdaForwardRole({
			prefix: "ThePrefix",
			pattern_name: "ThePatternName",
			bucket_name: "TheBucketName",
			bucket_path: "TheBucketPath"
		})).toMatchSnapshot();
	});
});
