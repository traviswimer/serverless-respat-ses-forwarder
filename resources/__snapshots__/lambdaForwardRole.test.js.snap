// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lambdaForwardRole() generates role 1`] = `
Object {
  "Properties": Object {
    "AssumeRolePolicyDocument": Object {
      "Statement": Array [
        Object {
          "Action": "sts:AssumeRole",
          "Effect": "Allow",
          "Principal": Object {
            "Service": "lambda.amazonaws.com",
          },
        },
      ],
      "Version": "2012-10-17",
    },
    "MaxSessionDuration": 3600,
    "Path": "TheBucketPath",
    "Policies": Array [
      Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "arn:aws:logs:*:*:*",
            },
            Object {
              "Action": "ses:SendRawEmail",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "s3:GetObject",
                "s3:PutObject",
                "s3:*",
              ],
              "Effect": "Allow",
              "Resource": "arn:aws:s3:::TheBucketName/*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ThePrefix-ThePatternName-role-policy",
      },
    ],
    "RoleName": "ThePrefix-ThePatternName-role",
  },
  "Type": "AWS::IAM::Role",
}
`;
